packages/framework/pr_tools/common.bash

```diff
@@ -193,23 +284,110 @@ function get_md5sum() {
 
 
 
+# get_python_packages()
+# Install required Python packages using pip.
+# installs: configparser, mock, pytest, pytest-cov
 #
-# Install Python pacakges using pip
-#
-# - @param1 pip_exe - the pip binary to use, i.e., pip3.
-#
+# param1: pip_exe - the pip binary to use, i.e., pip3.
 function get_python_packages() {
     local pip_exe=${1:?}
 
-    echo -e "--- Pip   : ${pip_exe:?}"
+    message_std "PRDriver> " "pip executable: ${pip_exe:?}"
 
     pip_args=(
         configparser
         mock
         pytest
         pytest-cov
     )
-    echo -e "--- ${pip_exe:?} install --user ${pip_args[@]}"
-    ${pip_exe:?} install --user ${pip_args[@]}
+    message_std "PRDriver> " ""
+    # ${pip_exe:?} install --user ${pip_args[@]}
+    execute_command "${pip_exe:?} install --user ${pip_args[@]}"
+}
+
+
+# executable_exists()
+# - Determines if a file exists and is executable.
+# param1: executable (with path if necessary)
+function executable_exists()
+{
+    local cmd=${1:?}
+    local output=1
+    if [ ! command -v ${cmd:?} &> /dev/null ]; then
+        output=0
+    fi
+    echo ${output:?}
```

Comment:
```suggestion
    return ${output:?}
```

Reply:
I am unsure what the use-case of this function is. If the caller relies on checking `$?`, we should use return rather than echo.